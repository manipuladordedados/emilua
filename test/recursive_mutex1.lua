local recursive_mutex = require 'recursive_mutex'
local time = require 'time'

local m = recursive_mutex.new()

m:lock()
m:lock()
print('1')

local f = spawn(function()
    m:lock()
    print('2')
    m:unlock()
end)

time.sleep(0.1)
m:unlock()
print('1b')

time.sleep(0.1)
m:unlock()

f:join()
