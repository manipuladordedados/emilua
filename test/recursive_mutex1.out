1
1b
2
